timeless/restaurants/table_shapes/views.py

```diff
@@ -49,23 +49,27 @@ def create():
 
 @bp.route("/edit/<int:id>", methods=("GET", "POST"))
 def edit(id):
-    """ Edit table shape with id
-    @todo #162:30min Implement edit() method of TableShape model and
-     update template. Use TableShapeForm for that, see how in works in
-     create function. Check and update html template if it's needed.
-    """
-    if request.method == "POST":
-        flash("Edit not yet implemented")
+    """Edit table shape with id"""
+    table = models.TableShape.query.get(id)
+    if not table:
+        return redirect(url_for("tables.list"))
+
+    form = forms.TableShapeForm(request.form, instance=table)
+    if request.method == "POST" and form.validate():
+        form.save()
+        return redirect(url_for("table_shape.list"))
 
-    form = forms.TableShapeForm()
+    form = forms.TableShapeForm(instance=table)
```

Comment:
@krzyk can you name this variable to ``edit`` or something more suggestive? It's the form that we display to the user to fill out, unlike the above one which is the completed ``form``.

Reply:
@krzyk also, can you put this GET logic inside an ``else``? It seems more understandable.

Reply:
@amihaiemil not quite, both forms are filled out, in case of GET it is the data from DB, in case of POST it is data from the form